### PR TITLE
Promotions index improvements

### PR DESCRIPTION
**Pull request by mamhoff**
Allows searching by lane, customer label and active date ("Which promotions will be active Wednesday next week?").

Also speeds up display of the index page for promotions with a very high usage count and no usage limit.

### CODE REVIEW

File: app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<table class="index">
+    <thead>
```
Comment (tvdeyen) [inline app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:2]: indentation off

File: app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb [hunk 2]
```diff
@@ -25,7 +25,7 @@
           </td>
           <td>
             <% if promotion.usage_limit.nil? %>
-              ∞
+              <%= promotion.discounted_orders.exists? ? t(:say_yes, scope: :spree) : t(:say_no, scope: :spree) %>
```
Comment (tvdeyen) [inline app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:28]: I do not understand what a "yes" or "no" means in context of a `nil` usage_count? Shouldn't this show the count of discounted orders instead?

**Review by tvdeyen — COMMENTED** [review]
Some questions. Also, at least some filtering integration spec would be nice, no?

File: app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb (hunk 2 above)
Comment (mamhoff) [reply app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:28]: `usage_count` is never `nil` - it can be zero for unused orders, or some really large number for used ones.

Comment (mamhoff) [reply app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:28]: So rather than tell the user nothing, we can tell them whether their promotion has been used in any complete order.

Comment (tvdeyen) [reply app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:28]: why not display discounted orders count?

File: app/controllers/solidus_friendly_promotions/admin/promotions_controller.rb [hunk 3]
```diff
@@ -36,6 +36,7 @@ def collection
 
         @collection = super
         @search = @collection.ransack(params[:q])
+        @search.sorts = "name asc" if @search.sorts.empty?
```
Comment (tvdeyen) [inline app/controllers/solidus_friendly_promotions/admin/promotions_controller.rb:39]: Wouldn't be `created_at DESC` much more practicable?

File: app/views/solidus_friendly_promotions/admin/promotions/_table_filter.html.erb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        <div class="col-2">
+          <div class="field">
+            <%= label_tag :q_lane_eq, SolidusFriendlyPromotions::Promotion.human_attribute_name(:lane) %><br>
+            <%= f.select(:lane_eq, SolidusFriendlyPromotions::Promotion.lane_options, { include_blank: t(:all, scope: :spree) }, { class: 'custom-select fullwidth' }) %>
+          </div>
+        </div>
+
+      </div>
+
+      <div class="clearfix"></div>
```
Comment (tvdeyen) [inline app/views/solidus_friendly_promotions/admin/promotions/_table_filter.html.erb:62]: Why is this necessary?

File: app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            <%= promotion.usage_count %> / <%= promotion.usage_limit %>
+          <% end %>
+        <td>
+          <%= promotion.starts_at.to_fs(:long) if promotion.starts_at %>
+        </td>
+        <td>
+          <%= promotion.expires_at.to_fs(:long) if promotion.expires_at %>
+        </td>
+        <td>
+            <%= SolidusFriendlyPromotions::Promotion.human_enum_name(:lane, promotion.lane) %>
```
Comment (tvdeyen) [inline app/views/solidus_friendly_promotions/admin/promotions/_table.html.erb:39]: Sorry, but my white-space scanner.... 🤣

File: app/views/solidus_friendly_promotions/admin/promotions/_table_filter.html.erb (hunk 4 above)
Comment (tvdeyen) [reply app/views/solidus_friendly_promotions/admin/promotions/_table_filter.html.erb:62]: comment still applies

**Review by tvdeyen — APPROVED** [review]
